src/ocean/util/log/StatsReader.d

```diff
@@ -252,6 +253,11 @@ public class StatsLogReader
 
         foreach (line; this.lines)
         {
+            if (line.length == 0)
```

Comment:
It seems that it might be better to wrap `this.lines` into a small struct which provides `opApply` and just ignores empty lines, then to repeat this over and over?